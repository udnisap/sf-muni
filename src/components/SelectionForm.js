import React, { Component } from 'react';
import PropTypes from 'prop-types';
import { Well, FormGroup, ControlLabel, FormControl } from 'react-bootstrap';

class SelectionForm extends Component {
  render() {
    const { routes, onChange } = this.props;
    return (
      <div className="form">
        <Well>
          <h3> SF - Muni Map</h3>
          <p>
            Welcome to SF-Muni map live update map. This will update every 15 seconds with the latest locations of the Munis in SF.
            {' '}
          </p>
          <p>
            To Check a specific route, use the dropdown below.
          </p>
          <FormGroup controlId="formControlsSelectMultiple">
            <ControlLabel>Select a Route</ControlLabel>
            <FormControl
              componentClass="select"
              onChange={event => {
                onChange(event.target.value);
              }}
            >
              <option value="all">all</option>
              {routes.map(route => (
                <option key={route} value={route}> {route} </option>
              ))}
            </FormControl>
          </FormGroup>

        </Well>
      </div>
    );
  }
}
SelectionForm.propTypes = {
  routes: PropTypes.arrayOf(PropTypes.string).isRequired,
  onChange: PropTypes.func.isRequired
};

export default SelectionForm;
