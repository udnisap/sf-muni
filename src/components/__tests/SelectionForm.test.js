import React from 'react';
import ReactDOM from 'react-dom';
import { shallow, mount, render } from 'enzyme';
import { configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

configure({ adapter: new Adapter() });
import SelectionForm from '../SelectionForm';

const props = {
  routes: ['test1', 'test2'],
  onChange: jest.fn()
};

describe('Selection Form', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SelectionForm {...props} />, div);
  });

  it("should render 'all' without any items", () => {
    const emptyRoutes = { ...props, routes: [] };
    const comp = render(<SelectionForm {...props} />);
    expect(comp.find('option').length).toEqual(1); // 'all' option
  });

  it('should match snapshot', () => {
    const comp = shallow(<SelectionForm {...props} />);
    expect.toMatchSnapshot();
  });

  it.only('should call onChange when select is changed', () => {
    const comp = mount(<SelectionForm {...props} />);
    const selection = 'test';
    comp.find('select').simulate('change', { target: { value: selection } });
    expect(props.onChange).toBeCalledWith(selection);
  });
});
