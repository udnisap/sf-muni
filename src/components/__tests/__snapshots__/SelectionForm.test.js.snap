// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Selection Form should match snapshot 1`] = `
initialize {
  "0": Object {
    "attribs": Object {},
    "children": Array [
      Object {
        "attribs": Object {
          "class": "well",
        },
        "children": Array [
          Object {
            "attribs": Object {
              "class": "form-group",
            },
            "children": Array [
              Object {
                "attribs": Object {
                  "class": "control-label",
                  "for": "formControlsSelectMultiple",
                },
                "children": Array [
                  Object {
                    "data": "Multiple select",
                    "next": null,
                    "parent": [Circular],
                    "prev": null,
                    "type": "text",
                  },
                ],
                "name": "label",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": Object {
                  "attribs": Object {
                    "class": "form-control",
                    "id": "formControlsSelectMultiple",
                  },
                  "children": Array [
                    Object {
                      "attribs": Object {
                        "value": "all",
                      },
                      "children": Array [
                        Object {
                          "data": "all",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "value": "test1",
                        },
                        "children": Array [
                          Object {
                            "data": " test1 ",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": Object {
                          "attribs": Object {
                            "value": "test2",
                          },
                          "children": Array [
                            Object {
                              "data": " test2 ",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "option",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": null,
                          "parent": [Circular],
                          "prev": [Circular],
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "value": undefined,
                          },
                        },
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "value": "test1",
                      },
                      "children": Array [
                        Object {
                          "data": " test1 ",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "value": "test2",
                        },
                        "children": Array [
                          Object {
                            "data": " test2 ",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "value": "all",
                        },
                        "children": Array [
                          Object {
                            "data": "all",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    Object {
                      "attribs": Object {
                        "value": "test2",
                      },
                      "children": Array [
                        Object {
                          "data": " test2 ",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "value": "test1",
                        },
                        "children": Array [
                          Object {
                            "data": " test1 ",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": Object {
                          "attribs": Object {
                            "value": "all",
                          },
                          "children": Array [
                            Object {
                              "data": "all",
                              "next": null,
                              "parent": [Circular],
                              "prev": null,
                              "type": "text",
                            },
                          ],
                          "name": "option",
                          "namespace": "http://www.w3.org/1999/xhtml",
                          "next": [Circular],
                          "parent": [Circular],
                          "prev": null,
                          "type": "tag",
                          "x-attribsNamespace": Object {
                            "value": undefined,
                          },
                          "x-attribsPrefix": Object {
                            "value": undefined,
                          },
                        },
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                  ],
                  "name": "select",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": null,
                  "parent": [Circular],
                  "prev": [Circular],
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "id": undefined,
                  },
                },
                "parent": [Circular],
                "prev": null,
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "for": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "for": undefined,
                },
              },
              Object {
                "attribs": Object {
                  "class": "form-control",
                  "id": "formControlsSelectMultiple",
                },
                "children": Array [
                  Object {
                    "attribs": Object {
                      "value": "all",
                    },
                    "children": Array [
                      Object {
                        "data": "all",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "option",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "value": "test1",
                      },
                      "children": Array [
                        Object {
                          "data": " test1 ",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": Object {
                        "attribs": Object {
                          "value": "test2",
                        },
                        "children": Array [
                          Object {
                            "data": " test2 ",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": null,
                        "parent": [Circular],
                        "prev": [Circular],
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": null,
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "value": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "value": "test1",
                    },
                    "children": Array [
                      Object {
                        "data": " test1 ",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "option",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": Object {
                      "attribs": Object {
                        "value": "test2",
                      },
                      "children": Array [
                        Object {
                          "data": " test2 ",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": null,
                      "parent": [Circular],
                      "prev": [Circular],
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "value": "all",
                      },
                      "children": Array [
                        Object {
                          "data": "all",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": null,
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "value": undefined,
                    },
                  },
                  Object {
                    "attribs": Object {
                      "value": "test2",
                    },
                    "children": Array [
                      Object {
                        "data": " test2 ",
                        "next": null,
                        "parent": [Circular],
                        "prev": null,
                        "type": "text",
                      },
                    ],
                    "name": "option",
                    "namespace": "http://www.w3.org/1999/xhtml",
                    "next": null,
                    "parent": [Circular],
                    "prev": Object {
                      "attribs": Object {
                        "value": "test1",
                      },
                      "children": Array [
                        Object {
                          "data": " test1 ",
                          "next": null,
                          "parent": [Circular],
                          "prev": null,
                          "type": "text",
                        },
                      ],
                      "name": "option",
                      "namespace": "http://www.w3.org/1999/xhtml",
                      "next": [Circular],
                      "parent": [Circular],
                      "prev": Object {
                        "attribs": Object {
                          "value": "all",
                        },
                        "children": Array [
                          Object {
                            "data": "all",
                            "next": null,
                            "parent": [Circular],
                            "prev": null,
                            "type": "text",
                          },
                        ],
                        "name": "option",
                        "namespace": "http://www.w3.org/1999/xhtml",
                        "next": [Circular],
                        "parent": [Circular],
                        "prev": null,
                        "type": "tag",
                        "x-attribsNamespace": Object {
                          "value": undefined,
                        },
                        "x-attribsPrefix": Object {
                          "value": undefined,
                        },
                      },
                      "type": "tag",
                      "x-attribsNamespace": Object {
                        "value": undefined,
                      },
                      "x-attribsPrefix": Object {
                        "value": undefined,
                      },
                    },
                    "type": "tag",
                    "x-attribsNamespace": Object {
                      "value": undefined,
                    },
                    "x-attribsPrefix": Object {
                      "value": undefined,
                    },
                  },
                ],
                "name": "select",
                "namespace": "http://www.w3.org/1999/xhtml",
                "next": null,
                "parent": [Circular],
                "prev": Object {
                  "attribs": Object {
                    "class": "control-label",
                    "for": "formControlsSelectMultiple",
                  },
                  "children": Array [
                    Object {
                      "data": "Multiple select",
                      "next": null,
                      "parent": [Circular],
                      "prev": null,
                      "type": "text",
                    },
                  ],
                  "name": "label",
                  "namespace": "http://www.w3.org/1999/xhtml",
                  "next": [Circular],
                  "parent": [Circular],
                  "prev": null,
                  "type": "tag",
                  "x-attribsNamespace": Object {
                    "class": undefined,
                    "for": undefined,
                  },
                  "x-attribsPrefix": Object {
                    "class": undefined,
                    "for": undefined,
                  },
                },
                "type": "tag",
                "x-attribsNamespace": Object {
                  "class": undefined,
                  "id": undefined,
                },
                "x-attribsPrefix": Object {
                  "class": undefined,
                  "id": undefined,
                },
              },
            ],
            "name": "div",
            "namespace": "http://www.w3.org/1999/xhtml",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "tag",
            "x-attribsNamespace": Object {
              "class": undefined,
            },
            "x-attribsPrefix": Object {
              "class": undefined,
            },
          },
        ],
        "name": "div",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "div",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {},
    "x-attribsPrefix": Object {},
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
